### ISSUE TITLE
Terminal windows don't persist

### ISSUE THREAD

**Issue by mfilej** [akinsho/toggleterm.nvim#430]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I just installed the plugin and I have no custom configuration. I have `set hidden`.

It seems that every time I hide the terminal split, the shell process gets terminated (I verified this with a long-running process).

I recorded a demo (see below).

### Expected Behavior

From what I understood in the documentation, terminal sessions should be persisted by default.

### Steps To Reproduce

1. Open terminal with `:ToggleTerm`
2. Run any command, preferably a long-running process.
3. Close terminal with `:ToggleTerm`. If there is a long-running process from the previous step, you should see it was terminated.
4. Re-open terminal with `:ToggleTerm`. A new shell session is started, no output from the previous shell session is visible.

### Environment

```Markdown
- OS: macOS Ventura 13.3
- neovim version: 0.8.3
- Shell: zsh (also confirmed with fish)
```

### Anything else?

![CleanShot 2023-04-20 at 11 18 17](https://user-images.githubusercontent.com/3621/233321513-89befc1e-f61a-445a-952d-970355010654.gif)

Please forgive me if this is an obvious user error (I suspect it might be).

**Comment by akinsho** [comment]
Hi @mfilej,

So as you point out this is likely a configuration issue, as there have been previous issues on this in the past which all came down to user config issue.

Using the `minimal.init` lua below with nvim `0.8.3` I'm able to open and close toggleterm and the process continues to run.

```lua
local lazypath = vim.fn.stdpath('data') .. '/lazy/lazy.nvim'
vim.opt.runtimepath:prepend(lazypath)

require('lazy').setup({
  { 'akinsho/toggleterm.nvim' },
})

require('toggleterm').setup({
  open_mapping = [[<c-\>]],
})
```

Can you please more information about your environment, are you reproducing this issue with a minimal init.lua? are there plugins or settings in place that might be interfering. Can you try a minimal config with no other settings if it doesn't reproduce in a minimal environment then that is certainly a config issue and I'd suggest doing like a binary search of your config to narrow down the cause

**Comment by mfilej** [comment]
Thanks, I should've thought of trying with a minimal config. Binary search is clearly the way to go, but I decided not to go down that road because in the meantime I stumbled upon some neovim `:terminal` limitations that prevent me from using terminal splits like I wanted to (example: https://github.com/neovim/neovim/issues/2514).

Still, this is a very cool project and definitely the best way to use terminal splits in my opinion, so I'm looking forward to getting back to it in the future.
